extensions/wikia/AdEngine/src/templates/big-fancy-ad-above-mobile-config.js

```diff
@@ -1,73 +1,93 @@
+		const height = aspectRatio ? viewPortWidth / aspectRatio : iframe.contentWindow.document.body.offsetHeight;
 
-function runOnReady(iframe, params, mercuryListener) {
-	const onResize = (aspectRatio) => {
-			adjustPadding(iframe, aspectRatio);
-		},
-		page = document.getElementsByClassName('application-wrapper')[0];
+		this.adsModule.setSiteHeadOffset(height);
+	},
 
-	page.classList.add('bfaa-template');
-	adjustPadding(iframe, params.aspectRatio);
-	window.addEventListener('resize', onResize.bind(null, params.aspectRatio));
+	onReady(iframe) {
+		const onResize = () => {
+			adjustPadding(iframe, this.params);
+		};
+		const page = document.querySelector('.application-wrapper');
 
-	if (mercuryListener) {
-		mercuryListener.onPageChange(() => {
-			page.classList.remove('bfaa-template');
-			document.body.classList.remove('vuap-loaded');
-			document.body.classList.remove('has-bfaa');
-			document.body.style.paddingTop = '';
-			adsModule.setSiteHeadOffset(0);
-			window.removeEventListener('resize', onResize);
-		});
-	}
-}
+		page.classList.add('bfaa-template');
+		this.adjustPadding(iframe, this.slotParams);
+		window.addEventListener('resize', onResize);
 
-export function getConfig(mercuryListener) {
-	let slotElement = null;
-	let navbarElement = null;
+		if (this.mercuryListener) {
+			this.mercuryListener.onPageChange(() => {
+				page.classList.remove('bfaa-template');
+				document.body.classList.remove('vuap-loaded');
+				document.body.classList.remove('has-bfaa');
+				document.body.style.paddingTop = '';
+				this.adsModule.setSiteHeadOffset(0);
+				window.removeEventListener('resize', onResize);
+			});
+		}
+	},
 
-	return {
-		slotsToEnable: [
-			'MOBILE_IN_CONTENT',
-			'MOBILE_PREFOOTER',
-			'MOBILE_BOTTOM_LEADERBOARD'
-		],
-		onInit(adSlot, params) {
-			adsModule = window.Mercury.Modules.Ads.getInstance();
+	onInit(adSlot, params) {
+		this.adSlot = adSlot;
+		this.slotParams = slotParams;
+		this.adsModule = window.Mercury.Modules.Ads.getInstance();
+		this.navbarElement = document.querySelector('.site-head-container .site-head');
 
-			context.set(`slots.${adSlot.getSlotName()}.options.isVideoMegaEnabled`, params.isVideoMegaEnabled);
+		const wrapper = document.querySelector('.mobile-top-leaderboard');
 
-			slotTweaker.onReady(adSlot).then((iframe) => runOnReady(iframe, params, mercuryListener));
+		context.set(`slots.${adSlot.getSlotName()}.options.isVideoMegaEnabled`, params.isVideoMegaEnabled);
+		wrapper.style.opacity = '0';
+		slotTweaker.onReady(adSlot).then((iframe) => {
+			wrapper.style.opacity = '';
+			this.runOnReady(iframe);
+		});
 
-			const wrapper = document.getElementsByClassName('mobile-top-leaderboard')[0];
+		if (this.adsModule.hideSmartBanner) {
+			this.adsModule.hideSmartBanner();
+		}
+	},
 
-			slotElement = adSlot.getElement();
-			navbarElement = document.querySelector('.site-head-container .site-head');
+	onUnstickBfaaCallback(adSlot) {
+		const slideOutDuration = 600;
```

Comment:
Animation time is defined by ad-products and cannot be different between platforms (animation time is hardcoded in ap-products css styles). Please at least use a constant for this timing. In an ideal scenario, we could trigger these animations from ad-products and only handle navbar here.